lib-gphotos/client.go

```diff
@@ -102,29 +104,49 @@ func (client *Client) UploadFile(filePath string, pAlbumID ...string) (*photosli
 		return nil, stacktrace.Propagate(err, "failed getting uploadToken for %s", filename)
 	}
 
-	batchResponse, err := client.MediaItems.BatchCreate(&photoslibrary.BatchCreateMediaItemsRequest{
-		AlbumId: albumID,
-		NewMediaItems: []*photoslibrary.NewMediaItem{
-			&photoslibrary.NewMediaItem{
-				Description:     filename,
-				SimpleMediaItem: &photoslibrary.SimpleMediaItem{UploadToken: uploadToken},
+	retry := true
+	retryCount := 0
+	for retry != false {
+		retry = false
+		batchResponse, err := client.MediaItems.BatchCreate(&photoslibrary.BatchCreateMediaItemsRequest{
+			AlbumId: albumID,
+			NewMediaItems: []*photoslibrary.NewMediaItem{
+				&photoslibrary.NewMediaItem{
+					Description:     filename,
+					SimpleMediaItem: &photoslibrary.SimpleMediaItem{UploadToken: uploadToken},
+				},
 			},
-		},
-	}).Do()
-	if err != nil {
-		return nil, stacktrace.Propagate(err, "failed adding media %s", filename)
-	}
+		}).Do()
+		if err != nil {
+			// handle rate limit error by sleeping and retrying
+			if strings.HasPrefix(err.Error(), "googleapi: Error 429") {
+				log.Printf("Rate limit reached, sleeping for 10 seconds...")
+				time.Sleep(10 * time.Second)
+				retry = true
+				retryCount++
+				continue
+			} else if retryCount < 3 {
+				log.Printf("Error during upload, sleeping for 10 seconds before retrying...")
+				time.Sleep(10 * time.Second)
```

Comment:
I'm not sure we need this second `else if`. I think we could change the `for retry != false {` for something like:
```
for retry != false && retryCount < 3 {
```
and remove this code

Reply:
we want to handle errors other than rate limit differently than other errors so I think this should still stay like this.